comms/examples/stress/service.rs

```diff
@@ -389,6 +435,40 @@ async fn start_responder_protocol(peer: NodeId, mut substream: Substream) -> Res
                 num_loops * 100
             );
         },
+        StressProtocolKind::BurstSend => {
+            let mut received = vec![];
+            let start = Instant::now();
+
+            for _i in 0..(protocol.num_messages / protocol.burst_size) {
+                while let Some(Ok(msg)) = framed.next().await {
+                    received.push(decode_msg(msg));
+                }
+            }
```

Comment:
`framed.next().await` will not return None after a burst. It will just `await` the next message. `next()` returns None once the stream has ended (e.g. disconnected substream). So your while loop will read all the messages, and subsequent loops will exit immediately (including the while loop below). 

  ```suggestion
                while let Some(Ok(msg)) = framed.next().await {
                    received.push(decode_msg(msg));
                }
            
```

Reply:
Thanks, did not realize.